src/pkg/repository/dao/dao.go

```diff
@@ -0,0 +1,38 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dao
+
+import (
+	"context"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/internal/orm"
+)
+
+// DAO is the data access object interface for repository
+type DAO interface {
```

Comment:
The name `DAO` we used here is too general

Reply:
`repository/dao/dao.go` -> `repository/dao/repository.go`

type Repository interface {}

When using, call `dao.Repository`?